lib/tesla/builder.ex

```diff
@@ -169,7 +169,9 @@ defmodule Tesla.Builder do
     end
   end
 
-  def client(pre, post), do: %Tesla.Client{pre: runtime(pre), post: runtime(post)}
+  def client(pre, post, adapter \\ nil) do
```

Comment:
I'd change it to something like:
```elixir
def client(pre, post) do
  %Tesla.Client{pre: runtime(pre), post: runtime(post)}
end

def client(pre, post, adapter) do
  %Tesla.Client{pre: runtime(pre), post: runtime(post), adapter: runtime(adapter)}
end
```
and remove this line: https://github.com/teamon/tesla/pull/244/files#diff-a2c274aba9c954a50446941dc12ebd88R207

Otherwise `nil` will be allowed as the list of middlewares.